src/main/java/org/dormitory/autobotsoccub/user/UserPool.java

```diff
@@ -0,0 +1,67 @@
+package org.dormitory.autobotsoccub.user;
+
+import lombok.Getter;
+import org.springframework.stereotype.Component;
+import org.telegram.telegrambots.api.objects.User;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Component
+public class UserPool {
+
+    private static final int POOL_SIZE = 4;
+    public static long POOL_REQUEST_TIMEOUT = TimeUnit.MINUTES.toMillis(15);
+    private long lastPoolUpdate;
+
+    @Getter private Map<Integer, User> userPool = new HashMap<>();
```

Comment:
- Let's think of better encapsulation/synchronization.
- Take existing expirable map implementation so we wouldn't rely on last timestamp and so on